### PR TITLE
feat: add case insensitive user search

### PR DESCRIPTION
**Pull request by ihsanguldur**
## Summary

Add case insensitive search for users.

## Testing

UT and Locally Tested.

### CODE REVIEW

**Comment by wangsijie** [comment]
hi @ihsanguldur, overall looks good, would you mind checking the lint issues?

**Comment by ihsanguldur** [comment]
> hi @ihsanguldur, overall looks good, would you mind checking the lint issues?

i will fix it as soon as possible.

File: packages/core/src/queries/user.ts [hunk 1]
```diff
@@ -84,44 +84,60 @@ export const hasUserWithIdentity = async (target: string, userId: string) =>
...
+    return sql`${sql.join(
+        searchFields.map((filedName) => sql`${filedName} like ${'%' + search + '%'}`),
+        sql` or `
+    )}`;
+  }
+
+  return sql`${sql.join(
+      searchFields.map((filedName) => sql`${filedName} ilike ${'%' + search + '%'}`),
+      sql` or `
+  )}`;
```
Comment (gao-sun) [inline packages/core/src/queries/user.ts:100]:
```suggestion
  return sql`${sql.join(
      searchFields.map((filedName) => sql`${filedName} ${isCaseSensitive ? sql`like` : sql`ilike`} ${'%' + search + '%'}`),
      sql` or `
  )}`;
```

File: packages/core/src/queries/user.ts [hunk 2]
```diff
@@ -84,44 +84,60 @@ export const hasUserWithIdentity = async (target: string, userId: string) =>
     `
   );
 
-const buildUserSearchConditionSql = (search: string) => {
+const buildUserSearchConditionSql = (search: string, isSensitive?: boolean) => {
```
Comment (gao-sun) [inline packages/core/src/queries/user.ts:87]:
```suggestion
const buildUserSearchConditionSql = (search: string, isCaseSensitive = false) => {
```

File: packages/core/src/queries/user.ts [hunk 3]
```diff
@@ -84,44 +84,60 @@ export const hasUserWithIdentity = async (target: string, userId: string) =>
...
-const buildUserConditions = (search?: string, hideAdminUser?: boolean) => {
+const buildUserConditions = (search?: string, hideAdminUser?: boolean, isSensitive?: boolean) => {
   if (hideAdminUser) {
     return sql`
       where not (${fields.roleNames}::jsonb?${UserRole.Admin})
-      ${conditionalSql(search, (search) => sql`and (${buildUserSearchConditionSql(search)})`)}
+      ${conditionalSql(
+          search,
+          (search) => sql`and (${buildUserSearchConditionSql(search, isSensitive)})`
+      )}
```
Comment (gao-sun) [inline packages/core/src/queries/user.ts:110]: @wangsijie this part can be refactored as above

File: packages/core/src/routes/admin-user.ts [hunk 4]
```diff
@@ -26,19 +26,24 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
...
     '/users',
     koaPagination(),
-    koaGuard({
-      query: object({ search: string().optional(), hideAdminUser: literal('true').optional() }),
-    }),
+      koaGuard({
+        query: object({
+          search: string().optional(),
+          hideAdminUser: literal('true').optional(),
+          isSensitive: z.enum(['true', 'false']).default('true'),
```
Comment (gao-sun) [inline packages/core/src/routes/admin-user.ts:33]:
```suggestion
          isCaseSensitive: z.enum(['true', 'false']).default('true'),
```

File: packages/core/src/queries/user.ts (hunk 2 above)
Comment (gao-sun) [reply packages/core/src/queries/user.ts:87]: or `isCaseSensitive` can be required

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/core/src/queries/user.test.ts [hunk 5]
```diff
@@ -278,16 +278,34 @@ describe('user query', () => {
     await expect(countUsers(search, true)).resolves.toEqual(dbvalue);
   });
 
+  it('countUsers with isSensitive', async () => {
```
Comment (charIeszhao) [inline packages/core/src/queries/user.test.ts:281]: nit-picking:
```suggestion
  it('countUsers with isCaseSensitive', async () => {
```

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/core/src/queries/user.test.ts [hunk 6]
```diff
@@ -342,6 +360,37 @@ describe('user query', () => {
     await expect(findUsers(limit, offset, search, true)).resolves.toEqual([dbvalue]);
   });
 
+  it('findUsers with isSensitive', async () => {
```
Comment (charIeszhao) [inline packages/core/src/queries/user.test.ts:363]:
```suggestion
  it('findUsers with isCaseSensitive', async () => {
```

File: packages/core/src/routes/admin-user.ts [hunk 7]
```diff
@@ -27,18 +27,23 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
     '/users',
     koaPagination(),
     koaGuard({
-      query: object({ search: string().optional(), hideAdminUser: literal('true').optional() }),
+      query: object({
+        search: string().optional(),
+        hideAdminUser: literal('true').optional(),
+        isCaseSensitive: z.enum(['true', 'false']).default('true'),
```
Comment (charIeszhao) [inline packages/core/src/routes/admin-user.ts:33]: Should set default to `false`, IMO

Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts:33]: Can align this guard with `hideAdminUser`: 

```ts
isCaseSensitive: literal('true').optional(),
```

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/routes/admin-user.ts (hunk 7 above)
Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts]: I've updated

**Pull request merged**